pkg/cmd/server/server.go

```diff
@@ -291,13 +305,50 @@ func (s *server) run() error {
 		}
 	}
 
-	if err := s.runControllers(config, backupStorageLocation); err != nil {
+	if err := s.runControllers(config, backupStorageLocation, defaultVolumeSnapshotLocations); err != nil {
 		return err
 	}
 
 	return nil
 }
 
+func (s *server) getDefaultVolumeSnapshotLocations() (map[string]*api.VolumeSnapshotLocation, error) {
+	providerDefaults := make(map[string]*api.VolumeSnapshotLocation)
+	if len(s.config.defaultVolumeSnapshotLocations) == 0 {
+		return providerDefaults, nil
+	}
+
+	volumeSnapshotLocations, err := s.arkClient.ArkV1().VolumeSnapshotLocations(s.namespace).List(metav1.ListOptions{})
+	if err != nil {
+		return providerDefaults, errors.WithStack(err)
+	}
+
+	providerLocations := make(map[string][]*api.VolumeSnapshotLocation)
+	for _, vsl := range volumeSnapshotLocations.Items {
+		providerLocations[vsl.Spec.Provider] = append(providerLocations[vsl.Spec.Provider], &vsl)
+	}
+
+	for provider, locations := range providerLocations {
+		defaultLocation, ok := s.config.defaultVolumeSnapshotLocations[provider]
+		if !ok {
+			return providerDefaults, errors.Errorf("missing provider %s. You must provide a default for every provider.", provider)
```

Comment:
I think we want to say something like `for every existing provider.`? Not sure this is the proper language but to indicate that it is not *all* providers, only those that have been specified.